src/System.Net.Http/src/System/Net/Http/MessageProcessingHandler.cs

```diff
@@ -12,6 +13,22 @@ namespace System.Net.Http
 {
     public abstract class MessageProcessingHandler : DelegatingHandler
     {
+        // Private class used to capture the SendAsync state in
+        // a closure, while simultaneously avoiding a tuple allocation.
+        private class SendState : TaskCompletionSource<HttpResponseMessage>
```

Comment:
Nit: As a matter of personal style, I would have defined this at the bottom of the containing class, but I don't think there's a guideline one way or the other.